tests/unit/tests_high_five_base.cpp

```diff
@@ -1319,6 +1319,96 @@ BOOST_AUTO_TEST_CASE(HighFiveCompounds) {
 }
 
 
+enum Position {
+    FIRST = 1,
+    SECOND = 2,
+    THIRD = 3,
+    LAST = -1,
+};
+
+enum class Direction: signed char {
+    FORWARD = 1,
+    BACKWARD = -1,
+    LEFT = -2,
+    RIGHT = 2,
+};
+
+// This is only for boost test
+std::ostream& operator<<(std::ostream& ost, const Direction& dir) {
+    ost << static_cast<int>(dir);
+    return ost;
+}
+
+namespace HighFive {
+    EnumType<Position> create_enum_position() {
+        return EnumType<Position>({{"FIRST", Position::FIRST},
+                                   {"SECOND", Position::SECOND},
+                                   {"THIRD", Position::THIRD},
+                                   {"LAST", Position::LAST}});
+    }
+
+    template<>
+    DataType create_datatype<Position>() {
```

Comment:
I know this might not be possible, but it would be awesome if we didnt have to add specializations to the HighFive namespace... I'm afraid this is hard for the common user...
maybe let's think a bit more about this before the release...

Reply:
I agree with you!

Reply:
What about providing the user a macro?

Reply:
```CREATE_ENUM(POSITION, FIRST, SECOND)```

Reply:
HIGHFIVE_REGISTER_TYPE (?)

Reply:
Not possible due to comma inside your argument macro

Reply:
Please update the example with the new syntax